tests/test_nightReporter.py

```diff
@@ -0,0 +1,122 @@
+# This file is part of rapid_analysis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test cases for the NightReporter."""
+
+import unittest
+import tempfile
+from unittest import mock
+
+import lsst.utils.tests
+
+import matplotlib as mpl
+mpl.use('Agg')
+
+from lsst.rapid.analysis.nightReport import NightReporter, loadReport, saveReport  # noqa: E402
+import lsst.rapid.analysis.butlerUtils as bu  # noqa: E402, N813
+import lsst.daf.butler as dafButler  # noqa: E402, N813
+
+
+class NightReporterTestCase(lsst.utils.tests.TestCase):
+    """A test case for testing the Night Reporter."""
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.butler = dafButler.Butler('LATISS', instrument='LATISS', collections=['LATISS/raw/all'])
+        except FileNotFoundError:
+            raise unittest.SkipTest("Skipping tests that require the LATISS butler repo.")
+
+        cls.dayObs = 20200316  # has 213 images with 3 different stars
+        cls.seqNums = bu.getSeqNumsForDayObs(cls.butler, cls.dayObs)
+        cls.nImages = len(cls.seqNums)
+
+        # TODO: DM-XXXXX remove 'NCSA' once we're using RFC-811 stuff
+        cls.reporter = NightReporter('NCSA', cls.dayObs)  # takes about 29s
+
+    def test_loadAndSave(self):
+        writeDir = tempfile.mkdtemp()
+        saveReport(self.reporter, writeDir)
+        loaded = loadReport(writeDir, self.dayObs)
+        self.assertIsInstance(loaded, lsst.rapid.analysis.nightReport.NightReporter)
+
+    def test_printObsTable(self):
+        with mock.patch('sys.stdout') as fake_stdout:
+            self.reporter.printObsTable()
+
+        # newline for each row plus header
+        self.assertEqual(len(fake_stdout.mock_calls), 2*(self.nImages+1))
+
+        tailNumber = 20
+        with mock.patch('sys.stdout') as fake_stdout:
+            self.reporter.printObsTable(tailNumber=tailNumber)
+        self.assertEqual(len(fake_stdout.mock_calls), 2*(tailNumber+1))
+
+    def test_plotPerObjectAirMass(self):
+        # We assume matplotlib is making plots, so just check that these
+        # don't crash.
```

Comment:
This kind of comment should be in a docstring for each test that behaves that way.